dashboard/app/models/user_profile.rb

```diff
@@ -58,11 +60,15 @@ def get_other_user_ids
   end
 
   # @param alternate_user_id [Integer] an alternate user_id for the user.
-  # TODO(asher): Determine whether we should validate the existence of an
-  #   account with user_id of alternate_user_id.
   def add_other_user_id(alternate_user_id)
     existing_other_user_ids = get_other_user_ids
+
+    # Exit early if the ID does not make sense: the ID is already an alternate,
+    # the ID is invalid, or the ID is the user ID for this user.
     return if existing_other_user_ids.include? alternate_user_id
+    return unless User.find_by_id(alternate_user_id)
+    return if alternate_user_id == self.user.id
```

Comment:
Since these rely on `user`, it's probably worth adding a `validates_presence_of :user`